Some changes for AI Kit/HAT+ (#3966)
raspberrypi/documentation: documentation/asciidoc/computers/ai/getting-started.adoc

Part Number: HM21LB1C2LAE
Product Name: HAILO-8L AI ACC M.2 B+M KEY MODULE EXT TMP
----

> NOTE: The AI HAT+ devices may show `<N/A>` for `Serial Number`, `Part Number` and `Product Name` above. This is expected, and will not affect device functionality.

Reviewer: ```suggestion

NOTE: AI HAT+ devices may show `<N/A>` for `Serial Number`, `Part Number` and `Product Name`. This is expected, and does not impact functionality.
```

The following demos use xref:camera_software.adoc#rpicam-hello[`rpicam-hello`], which by default displays a preview window. However, you can use other `rpicam-apps` instead, including xref:camera_software.adoc#rpicam-vid[`rpicam-vid`] and xref:camera_software.adoc#rpicam-still[`rpicam-still`]. You may need to add or modify some command line options to make the demo commands compatible with alternative applications.

> To begin, ensure you have the latest `rpicam-apps` software package installed with:

Reviewer: ```suggestion
To begin, run the following command to install the latest `rpicam-apps` software package:
```

[source,console]
----

> $ sudo apt install rpicam-apps

Reviewer 2: Should we suggest that users run an `sudo apt update` first?

Reviewer: probably!

Reviewer: ```suggestion
$ sudo apt update && sudo apt install rpicam-apps
```
This should be enough, no?

[source,console]
----
$ rpicam-hello -t 0 --post-process-file /usr/share/rpi-camera-assets/hailo_yolov8_pose.json
----

=== Alternative Package Versions

> The AI Kit/HAT+ devices will not function if there is a version mismatch between the Hailo software packages and device drivers. In addition, Hailo's neural network tooling may also require particular a version for the generated model file. At the time of writing, the latest package version is 4.19, but if earlier versions are required, the following steps should be followed:

Reviewer: ```suggestion
The AI Kit and AI HAT+ do not function if there is a version mismatch between the Hailo software packages and device drivers. In addition, Hailo's neural network tooling may also require particular a version for the generated model file. If you require an specific version, complete the following steps to install the proper versions of all of the dependencies:
```

Reviewer 2: "an specific" -> "a specific" :stuck_out_tongue_winking_eye:

> The AI Kit/HAT+ devices will not function if there is a version mismatch between the Hailo software packages and device drivers. In addition, Hailo's neural network tooling may also require particular a version for the generated model file. At the time of writing, the latest package version is 4.19, but if earlier versions are required, the following steps should be followed:

Reviewer 2: "require particular a version" -> "require a particular version"

> The AI Kit/HAT+ devices will not function if there is a version mismatch between the Hailo software packages and device drivers. In addition, Hailo's neural network tooling may also require particular a version for the generated model file. At the time of writing, the latest package version is 4.19, but if earlier versions are required, the following steps should be followed:

Reviewer 2: Is there a reason why a user might want to explicitly use an older version, rather than just using whatever gets auto-installed by an `apt update`?

Author: Some older models won't work with the latest packages.

Reviewer 2: Aha, thanks.

> . If you have previously used `apt-mark` to hold any of the relevant packages, you may need to unhold them:

Reviewer 2: Is this something that a typical user is _likely_ to have done?

Author: Only if they have an explicit reason to do so, e.g. running older models.

+
[source,console]
----
$ sudo apt-mark unhold hailo-tappas-core hailort hailo-dkms
----

> . Install the required version of the software packages

Reviewer: ```suggestion
. Install the required version of the software packages:
```

Nitpick: as in humans, colons are important 💩

+
[tabs%sync]

> ====
> v4.18::
> +
> [source,console]
> ----
> $ sudo apt install hailo-tappas-core=3.29.1 hailort=4.18.0 hailo-dkms=4.18.0-2
> $ sudo apt-mark hold hailo-tappas-core hailort hailo-dkms
> ----
> 
> v4.17::
> +
> [source,console]
> ----
> $ sudo apt install hailo-tappas-core-3.28.2 hailort=4.17.0 hailo-dkms=4.17.0-1
> $ sudo apt-mark hold hailo-tappas-core hailort hailo-dkms
> ----
> ====

Reviewer: ```suggestion
======
v4.18::
To install version 4.18 of Hailo's neural network tooling:
+
+
[source,console]
----
$ sudo apt install hailo-tappas-core=3.29.1 hailort=4.18.0 hailo-dkms=4.18.0-2
----
+
[source,console]
----
$ sudo apt-mark hold hailo-tappas-core hailort hailo-dkms
----

v4.17::
To install version 4.17 of Hailo's neural network tooling:
+
[source,console]
----
$ sudo apt install hailo-tappas-core-3.28.2 hailort=4.17.0 hailo-dkms=4.17.0-1
----
+
[source,console]
----
$ sudo apt-mark hold hailo-tappas-core hailort hailo-dkms
----
======
```

Reviewer: You wouldn't know it without testing, but console snippets are awkward to copy/paste when we smush them together into a single code block. So I tend to split them up!

[source,console]
----
$ rpicam-hello -t 0 --post-process-file /usr/share/rpi-camera-assets/hailo_yolov8_pose.json
----

=== Alternative Package Versions

> The AI Kit and AI HAT+ do not function if there is a version mismatch between the Hailo software packages and device drivers. In addition, Hailo's neural network tooling may also require particular a version for the generated model file. If you require an specific version, complete the following steps to install the proper versions of all of the dependencies:

Reviewer: ```suggestion
The AI Kit and AI HAT+ do not function if there is a version mismatch between the Hailo software packages and device drivers. In addition, Hailo's neural network tooling may require a particular version for generated model files. If you require an specific version, complete the following steps to install the proper versions of all of the dependencies:
```
